retry.go

```diff
@@ -70,13 +71,11 @@ func (c Client) newRetryTimer(maxRetry int, unit time.Duration, cap time.Duratio
 		defer close(attemptCh)
 		for i := 0; i < maxRetry; i++ {
 			select {
-			// Attempts start from 1.
 			case attemptCh <- i + 1:
-			case <-doneCh:
-				// Stop the routine.
+			case <-time.After(exponentialBackoffWait(i)):
```

Comment:
We can send wrong attempt number here after some timeout.

The better way is:
```
select {
case attemptCh <- i+1:
case <-ctx.Done():
   return 
}

case {
case <-time.After(exponentialBackoffWait(i)):
case <-ctx.Done():
   return
}
```

Reply:
This is the ideal fix